### PR TITLE
Set default A records on user request

### PR DESCRIPTION
**Pull request by delputnam**
#### Changes proposed in this Pull Request

* Add the capability to set the default A records and remove an existing root A/AAAA records.

#### Testing instructions

Set non-default A and AAAA records on a domain.
You can also set the pointsToWpcom value for the test domain to `false` to make it easier to test.

In the DNS menu, use the action button at the top of page and select the option to reset the default A records.
Make sure that you get the success message and that the page is re-rendered with the new records.
Also use the patch instructions to fake an error and make sure that the error message is shown and the rendered records are unchanged.

Also make sure that the deleted records and the added records are in the same changeset.

### CODE REVIEW

File: client/my-sites/domains/domain-management/dns/dns-add-new-record-button.tsx [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import { Button } from '@automattic/components';
 import { useI18n } from '@wordpress/react-i18n';
 import { domainManagementDnsAddRecord } from 'calypso/my-sites/domains/paths';
-import './dns-breadcrumb-button.scss';
```
Comment (delputnam) [inline client/my-sites/domains/domain-management/dns/dns-add-new-record-button.tsx:4]: This is handled by the breadcrumb styles now.

File: client/my-sites/domains/domain-management/dns/dns-menu-options-button.tsx [hunk 2]
```diff
@@ -3,49 +3,56 @@ import { Button } from '@automattic/components';
...
 	const toggleMenu = useCallback( () => {
 		setMenuVisible( ! isMenuVisible );
 	}, [ isMenuVisible ] );
 
 	const closeMenu = useCallback( () => setMenuVisible( false ), [] );
 
-	const restoreDefaultRecords = useCallback( async () => {
-		const wpcomDomain = wpcom.domain( domain );
+	const getRecordsToRemove = useCallback( () => {
+		const dnsRecords = dns?.records ?? [];
```
Comment (rafaelgallani) [inline client/my-sites/domains/domain-management/dns/dns-menu-options-button.tsx]: Minor:
Could the `dns` prop be falsy? 
If so, then I think we should make it optional on the `DnsMenuOptionsButtonProps` type.
If not, then we wouldn't need this optional check (`dns?.records`).

**Review by rafaelgallani — APPROVED** [review]
LGTM! I left a single minor comment.

Tests:

#### Successful restore:

https://user-images.githubusercontent.com/18705930/142028149-c9315f9e-e772-4483-a9fe-36a3b3b35636.mov

#### Simulating an error:

image

File: client/my-sites/domains/domain-management/dns/dns-menu-options-button.tsx (hunk 2 above)
Comment (delputnam) [reply client/my-sites/domains/domain-management/dns/dns-menu-options-button.tsx]: Good call! I'll remove the check.

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6978902

Hi @delputnam, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Default A records restored`
- `Failed to restore the default A records`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6978902).

**Pull request merged**